### ISSUE TITLE
date_created value for "customer created" web hook showing wrong time.

### ISSUE THREAD

**Issue by AbdulSamadKt** [woocommerce/woocommerce#22247]
- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue

## Steps to reproduce the issue

1. Use Melbourne timezone in Settings -> General 
2.  Create a web hook with customer created in WP REST API integration v2
3. Register a user through my account page and test the web hook http request. 

## Expected/actual behavior
When I register a user to the site, the web hook for customer created shows wrong **date_created** and **date_created_gmt**. 

I created a user in 11:51 am GMT. My site is on GMT + 11 timezone (Melbourne)
Here is the http response:
{
  "id": 6,
  "date_created": "2018-12-17T11:51:54",
  "date_created_gmt": "2018-12-17T00:51:54",
  "date_modified": null,
  "date_modified_gmt": null,
  "email": "test@test.com",
  "first_name": "",
  "last_name": "",
  "role": "customer",
  "username": "test",
  "billing": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "state": "",
    "postcode": "",
    "country": "",
    "email": "",
    "phone": ""
  },
  "shipping": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "state": "",
    "postcode": "",
    "country": ""
  },
  "is_paying_customer": false,
  "orders_count": 0,
  "total_spent": "0.00",
  "avatar_url": "http://2.gravatar.com/avatar/b642b4217b34b1e8d3bd915fc65c4452?s=96&d=mm&r=g",
  "meta_data": [
    {
      "id": 170,
      "key": "wc_last_active",
      "value": "1545004800"
    }
  ]
}

The date_created need to show local site time of registration instead of GMT time and date_created_gmt need to show GMT time of the user registration. 

When I test other web hooks like coupon and order create, those are working as expected. On those date_created shows the local site time and  date_created_gmt shows that local time converted to GMT

## Isolating the problem

- [ x] This bug happens with only WooCommerce plugin active
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>

`
### WordPress Environment ###

Home URL: http://woocommerce-148425-466749.cloudwaysapps.com
Site URL: http://woocommerce-148425-466749.cloudwaysapps.com
WC Version: 3.5.2
Log Directory Writable: ✔
WP Version: ❌ 4.9.9 - There is a newer version of WordPress available (5.0.1)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.10 (Debian)
PHP Version: 7.0.32-1~dotdeb+8.1 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 10 MB
PHP Time Limit: 60
PHP Max Input Vars: 2500
cURL Version: 7.38.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
MySQL Version: ❌ 5.5.62-0+deb8u1 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 10 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.6
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 39.08MB
Database Data Size: 33.07MB
Database Index Size: 6.01MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.09MB
wp_custom_checkout_url: Data: 0.02MB + Index: 0.00MB
wp_hidemysiteSecure: Data: 0.02MB + Index: 0.00MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_mobileassistant_accounts: Data: 0.02MB + Index: 0.02MB
wp_mobileassistant_devices: Data: 0.02MB + Index: 0.02MB
wp_mobileassistant_failed_login: Data: 0.02MB + Index: 0.00MB
wp_mobileassistant_push_settings: Data: 0.02MB + Index: 0.00MB
wp_mobileassistant_session_keys: Data: 0.02MB + Index: 0.00MB
wp_mobileassistant_users: Data: 0.02MB + Index: 0.02MB
wp_options: Data: 4.06MB + Index: 0.05MB
wp_pieregister_code: Data: 0.00MB + Index: 0.00MB
wp_pieregister_lockdowns: Data: 0.00MB + Index: 0.00MB
wp_pieregister_redirect_settings: Data: 0.00MB + Index: 0.00MB
wp_postmeta: Data: 24.47MB + Index: 4.02MB
wp_posts: Data: 3.45MB + Index: 0.77MB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB
wp_strong_views: Data: 0.02MB + Index: 0.00MB
wp_termmeta: Data: 0.08MB + Index: 0.09MB
wp_terms: Data: 0.08MB + Index: 0.09MB
wp_term_relationships: Data: 0.22MB + Index: 0.13MB
wp_term_taxonomy: Data: 0.09MB + Index: 0.14MB
wp_usermeta: Data: 0.02MB + Index: 0.03MB
wp_users: Data: 0.02MB + Index: 0.05MB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

attachment: 3412
page: 14
post: 2
product: 486
product_variation: 11
revision: 7
rooms: 1
scheduled-action: 2
shop_coupon: 6
shop_order: 5
wpm-testimonial: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 3.5.2

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: 
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #39 - /shop
Cart: #40 - /cart
Checkout: #41 - /checkout
My account: #42 - /my-account
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.3.5 – 2.4.2 is available
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by SiR-DanieL** [comment]
Reference 8896665-hc

**Comment by AbdulSamadKt** [comment]
Any updates?

**Comment by rrennick** [comment]
I tested this and confirmed this is an issue. My timezone is UTC-4:00. The `date_modified*` are the correct values for the time of user creation. 

```
    "date_created": "2018-12-05T14:12:48",
    "date_created_gmt": "2018-12-05T18:12:48",
    "date_modified": "2018-12-05T11:47:00",
    "date_modified_gmt": "2018-12-05T15:47:00",
```

**Comment by sarun007** [comment]
any update for this issue ??

**Comment by rodrigoprimo** [comment]
See #22269 for a work in progress fix to this issue.

**Comment by AbdulSamadKt** [comment]
@rrennick 

I did another test now at 5:18 GMT and I got below values from the customer created webhook. Do you think this is correct?
"date_created": "2019-05-23T05:18:19",
"date_created_gmt": "2019-05-22T19:18:19"

For more clarity, I have created another webhook now for coupon created and the result is below at the time 05:41 GMT and for me, it shows correctly.

"date_created": "2019-05-23T15:41:23",
"date_created_gmt": "2019-05-23T05:41:23"

So my doubt is why customer created webhook and coupon created webhook showing completely different times? ( both are tested in just 23 minutes difference)

**Comment by rrennick** [comment]
@AbdulSamadKt The steps in your original description can be used to recreate this issue?

**Comment by AbdulSamadKt** [comment]
@rrennick Yes. Same steps can use to recreate the issue.

**Comment by rrennick** [comment]
I just tested this with my dev install. 

POSTed to REST

```json
{"email":"rdoe@example.com","first_name":"ronnie","last_name":"Doe","username":"rondoe","password":"testpassword","billing_address":{"first_name":"Ron","last_name":"Doe","company":"","address_1":"969 Market","address_2":"","city":"San Francisco","state":"CA","postcode":"94103","country":"US","email":"rdoe@example.com","phone":"(555) 555-5555"}}
```

REST return data

```json
{
  "id": 10,
  "date_created": "2019-06-11T04:09:35",
  "date_created_gmt": "2019-06-10T18:09:35",
  "date_modified": "2019-06-11T04:09:35",
  "date_modified_gmt": "2019-06-10T18:09:35",
  "email": "rdoe@example.com",
  "first_name": "ronnie",
  "last_name": "Doe",
  "role": "customer",
  "username": "rondoe",
  "billing": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "postcode": "",
    "country": "",
    "state": "",
    "email": "",
    "phone": ""
  },
  "shipping": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "postcode": "",
    "country": "",
    "state": ""
  },
  "is_paying_customer": false,
  "avatar_url": "https://secure.gravatar.com/avatar/b30860a80581bd4d77f9add614d277c5?s=96&d=mm&r=g",
  "meta_data": [],
  "_links": {
    "self": [
      {
        "href": "https://ron.localhost/wp-json/wc/v3/customers/10"
      }
    ],
    "collection": [
      {
        "href": "https://ron.localhost/wp-json/wc/v3/customers"
      }
    ]
  }
}
```

Sent via [webhook](https://webhook.site/#!/78d9cb2d-54ea-4292-9daa-878e773586cc/e9e2207f-7d2d-4d38-bd88-507d818ef131/1)

```json
{
  "id": 10,
  "date_created": "2019-06-11T04:09:35",
  "date_created_gmt": "2019-06-10T18:09:35",
  "date_modified": "2019-06-11T04:09:35",
  "date_modified_gmt": "2019-06-10T18:09:35",
  "email": "rdoe@example.com",
  "first_name": "ronnie",
  "last_name": "Doe",
  "role": "customer",
  "username": "rondoe",
  "billing": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "postcode": "",
    "country": "",
    "state": "",
    "email": "",
    "phone": ""
  },
  "shipping": {
    "first_name": "",
    "last_name": "",
    "company": "",
    "address_1": "",
    "address_2": "",
    "city": "",
    "postcode": "",
    "country": "",
    "state": ""
  },
  "is_paying_customer": false,
  "avatar_url": "https://secure.gravatar.com/avatar/b30860a80581bd4d77f9add614d277c5?s=96&d=mm&r=g",
  "meta_data": []
}
```

**Comment by AbdulSamadKt** [comment]
On REST API, it is working correctly but on webhook, it is not.

**Comment by rrennick** [comment]
@AbdulSamadKt In the test above I had the same times on the webhook as was returned from the rest api. 

We need to be able to reproduce the issue to investigate/fix. If you could provide the steps for us to follow to reproduce that would help.

**Comment by sarun007** [comment]
Hi,

I have replicated the issue on latest woocommerce version 3.6.4. When new customer registering the date_created and  date_created_gmt time are showing wrong on webhook. For replicating the issue i have attached video for reference. We are using https://dev.oscato.com tool for webhook checking. The date_created time showing GMT time and date_created_gmt Time showing wrong time on webhook response.
https://embed.vidyard.com/share/qrxx6QcbnajkQTSPyrNuJh?

Please solve the issue ASAP. We are running only woocommerce and storefront theme so the issue related with woocommerce. 

Thanks Sarun

**Comment by rrennick** [comment]
@sarun007 Per my previous comment, can you provide instructions to reproduce this issue in our dev installs?

**Comment by sarun007** [comment]
@rrennick did you watched my video reference ? On that video i have provided the instructions to reproduce this issue.

**Comment by sarun007** [comment]
Hi,

Any update for the issue ??

**Comment by claudiosanches** [comment]
I just tested on WooCommerce 3.7 RC and works fine, but only in REST API v3, need to change the " API Version" in the webhook settings.
I'll include this patch for REST API v1 and v2. will be included into WooCommerce 3.7.1.

**Comment by claudiosanches** [comment]
I have created a patch for v2: https://github.com/woocommerce/woocommerce-rest-api/pull/10
Closing it since should be patched in 3.7.1.
